tests/hrex/test_hrex.py

```diff
@@ -40,3 +44,105 @@ def test_get_samples_by_iter_by_replica_invalid_args():
         get_samples_by_iter_by_replica([[1], [2, 3]], [[ReplicaIdx(0)], [ReplicaIdx(0)]])
     with pytest.raises(AssertionError):
         get_samples_by_iter_by_replica([[1, 2], [3, 4]], [[ReplicaIdx(0)], [ReplicaIdx(0), ReplicaIdx(1)]])
+
+
+def simulate_perfect_mixing_hrex(num_states: int, num_frames: int) -> List[List[int]]:
+    """assume every step of HREX perfectly mixed all replicas"""
+    rng = np.random.default_rng(num_states)
```

Comment:
nit: unexpectedly frozen random function -- consider accepting rng or seed as argument if needed for explicitness (or depending on global numpy random state as in the next 2 fxns)

(otherwise, this fxn will produce the same sequence when called multiple times, which may cause testing bugs if we attempt to run multiple trials)

Reply:
Fixed up in https://github.com/proteneer/timemachine/pull/1314/commits/68079aab2b6b8fdcef0f050a75313708516ba6a3